File: Ejercicio_1.py
import psycopg2

while True:

    try:

        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "12345abc",
            dbname = "Tarea_Preparatoria"
        )

        print("""
        1) Abrir programa
        2) Ver historial""")

        nn1 = int(input("Seleccione una opcion: "))

        if nn1==1:
            print("Introduzca su fecha de nacimiento")
            print("Ejemplo:   Dia: 5   Mes: 2  Año: 2000")


            dia1 = int(input("Dia: "))
            mes1 = int(input("Mes: "))
            year1 = int(input("Año: "))

            dia2 = 24  #coloque la fecha de hoy
            mes2 = 2   # pero se puede modificar para recibir la fecha de nacimiento y la fecha en la que se esta haciendo la consulta
            year2 = 2022

            Edad = year2-year1

            x1=0
            x2=0

            if mes1<mes2:
                print("Esta persona ya cumplio años")
                x1=Edad
                print("Esta persona tiene ", x1, " años")
                x2 = "Esta pesona ya cumplio años"

            elif (mes1==mes2) & (dia1<=dia2):
                print("Esta persona ya cumplio años")
                x1=Edad
                print("Esta persona tiene ", x1," años")
                x2 = "Esta persona ya cumplio años"
            
            elif (mes1>mes2):
                print("Esta persona aun no cumple años")
                x1=Edad-1
                print("Esta persona tiene ",x1, " años")
                x2 = "Esta persona aun no cumple años"
            elif (mes1==mes2)&(dia1>dia2):
                print("Esta persona aun no cumple años")
                x1=Edad-1
                print("Esta persona tiene ", x1," años")
                x2 = "Esta persona aun no cumple años"

            cursor = conexion.cursor()
            cursor.execute("INSERT INTO ejer1(dia,mes,ano,edad,celebro) VALUES(%s,%s,%s,%s,%s);",(dia1,mes1,year1,x1,x2))
            conexion.commit()
            cursor.close()
            conexion.close()


        elif nn1==2:
            cursor = conexion.cursor()
            SQL = "SELECT*FROM ejer1;"
            cursor.execute(SQL)
            valores = cursor.fetchall()
            print(valores)
            cursor.close()
            conexion.close()
    except:
        print("Error: Revise la informacion que ha proporcionado")


        

File: Ejercicio__1.m
clc;clear;

try
  pkg load database
  conn = pq_connect(setdbopts('dbname','Tarea_Preparatoria','host','localhost','port','5432','user','postgres','password','12345abc'));
	
  fprintf('Introduzca su edad \n');
  fprintf('Ejmplo : Dia: 15  Mes: 14 Año: 1998 \n ')
	dia1 = input('Ingrese el dia de su cumpleaños: ');
  mes1 = input('Ingrese el numero de su mes de nacimiento: ');
  year1 = input('Ingrese el año en que nacio: ');
  printf('Usted nacio el dia %i del mes %i y año %i \n', dia1, mes1, year1);
  edad=2022-year1;
  
  if (mes1<2)
    printf('Usted ya cumplio %i años ', edad);
    concat=strcat(' Usted nacio el dia ', num2str(dia1), ' del mes ', num2str(mes1),' y año ',num2str(year1));
	  concat1=strcat(' Ya cumplio ', num2str(edad), ' años');
    N = pq_exec_params(conn,"insert into ejer1(dia, mes,ano,edad,celebro) values($1,$2,$3,$4,$5);",{dia1,mes1,year1,edad,'Ya cumplio'});
  elseif (mes1==2)&(dia<=24)
    printf('Usted ya cumplio %i años ', edad);
    concat=strcat(' Usted nacio el dia ', num2str(dia1), ' del mes ', num2str(mes1),' y año ',num2str(year1));
	  concat1=strcat(' Ya cumplio ', num2str(edad), ' años');
    N = pq_exec_params(conn,"insert into ejer1(dia, mes,ano,edad,celebro) values($1,$2,$3,$4,$5);",{dia1,mes1,year1,edad,'Ya cumplio'});
  
  elseif (mes1>2)
    printf('Usted ya no ha cumplido %i años', edad);
    concat=strcat(' Usted nacio el dia ', num2str(dia1), ' del mes ', num2str(mes1),' y año ',num2str(year1));
	  concat1=strcat(' Ya cumplio ', num2str(edad), ' años');
    N = pq_exec_params(conn,"insert into ejer1(dia, mes,ano,edad,celebro) values($1,$2,$3,$4,$5);",{dia1,mes1,year1,edad,'Ya no ha cumplido'});
  
  elseif (mes1==24)&(dia1>24)
    printf('Usted ya no ha cumplido %i años', edad);
    concat=strcat(' Usted nacio el dia ', num2str(dia1), ' del mes ', num2str(mes1),' y año ',num2str(year1));
	  concat1=strcat(' Ya cumplio ', num2str(edad), ' años');
    N = pq_exec_params(conn,"insert into ejer1(dia, mes,ano,edad,celebro) values($1,$2,$3,$4,$5);",{dia1,mes1,year1,edad,'Ya no ha cumplido'});

  endif


catch
	printf("Ha ocurrido un error")
end_try_catch
